Addresses MYRIAD-171 and MYRIAD-190.  Bumped the version to demonstra…

DarinJ opened the pull request:
…te functionality of the build.  Need to upgrade the vagrant and docker images to 2.7.2.

WIP - Don't Merge just yet.

yufeldman commented on build.gradle:
```diff
@@ -53,7 +53,7 @@ subprojects {
     ext {
         mesosVer = "0.24.1"
-        hadoopVer = "2.7.0"
+        hadoopVer = "2.7.2"
```
We should have this as property probably so not to change build.gradle every time

DarinJ replied:
Sure

yufeldman commented on myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java:
```diff
@@ -92,12 +97,39 @@ public synchronized void storeMyriadState(StoreContext sc) throws Exception {
+  }
+
+  protected void reflectedUpdateFile(Path outputPath, byte[] data) throws InvocationTargetException, IllegalAccessException {
+    if (cachedUpdateFileMethod == null) {
```
Why do you need to check here for null - you already do it inside getCachedUpdateFileMethod() method

DarinJ replied:
After looking over this again, I think I can just initialize once in `InitInternal(...)` and drop the lazy eval business.  Solves the null checks much cleaner.

yufeldman commented:
two minor comments, otherwise looks good. +1

DarinJ commented:
@yufeldman I think this commit addresses all your concerns.  Thanks for the review, definitely better to initialize the method at startup than mess with synchronization and lazy initialization!

yufeldman commented:
LGTM. +1 (though it is not binding, since I am not a commiter)

sdaingade commented:
LGTM +1
